Working on problem3

paulaOliverosHollis opened the pull request:
Solved problem 3!

paulaOliverosHollis committed:
Solved problem 3.

paulaOliverosHollis committed:
Started to work on problem 4.

paulaOliverosHollis committed:
Getting latest from master.

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
             Console.WriteLine($"\n\tSolution: {sum}");
         }
 
+
```
Whitespace!

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+
         public static void Problem3()
         {
+            Console.WriteLine("\n\t3) What is the largest prime factor of the number 600851475143 ?");
```
Can you make this easier to read? ex: 600,851,475,143

zhollis21 replied:
Also you left a space between the number and the question mark.

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+            Console.WriteLine("\n\t3) What is the largest prime factor of the number 600851475143 ?");
+
+            int smallestPrimeNumber = 2;
+            long biggestPrimeNumber = 600851475143;
```
This isn't a prime number...

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+        private static bool IsItDivisible(int smallestPrimeNumber, long biggestPrimeNumber)
+        {
+            return biggestPrimeNumber % smallestPrimeNumber == 0;
+        }
```
Since these methods go with problem number 3, put them under problem 3, not problem 4

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+            int smallestPrimeNumber = 2;
+            long biggestPrimeNumber = 600851475143;
+
+            while (true)
```
Why is this not a for loop?

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+            {
+                while (IsItDivisible(smallestPrimeNumber, biggestPrimeNumber))
+                {
+                    biggestPrimeNumber /= smallestPrimeNumber;
```
Why are you overwriting your reference number?

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+
+            while (true)
+            {
+                while (IsItDivisible(smallestPrimeNumber, biggestPrimeNumber))
```
Why are you looping here?

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+                {
+                    break;
+                }
+                else
```
Really no point in putting an else here

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+        {
+            double squareRoot = Math.Sqrt(biggestPrimeNumber);
+
+
```
Whitespace

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+        }
+
+        //Helper method.
+        private static bool IsItPrime(long biggestPrimeNumber)
```
Why are you calling it biggestPrimeNumber? We don't even know if it's prime yet...

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+        //Helper method.
+        private static bool IsItPrime(long biggestPrimeNumber)
+        {
+            double squareRoot = Math.Sqrt(biggestPrimeNumber);
```
Because comparing a double takes more time, I would convert this to an int since you are comparing it with _i_ anyways.

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+            {
+                if (IsItDivisible(i, biggestPrimeNumber))
+                {
+                    return false;
```
This looks good. Nice job

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+        }
+
+        //Helper method.
+        private static bool IsItDivisible(int smallestPrimeNumber, long biggestPrimeNumber)
```
This is a generic helper method that you could use for any method, but your variable name are specific to problem 3 and don't even make sense in this context.

zhollis21 commented on Project-Euler/Project-Euler/Problems.cs:
```diff
@@ -55,9 +55,77 @@ public static void Problem2()
+            return biggestPrimeNumber % smallestPrimeNumber == 0;
+        }
+
+        //Helper method.
```
This is a bad comment

zhollis21 replied:
Also whitespace before helper

paulaOliverosHollis committed:
Fixed some variable naming issues and added comments.

I change a while  true loop for a for loop to make the code for readable and reliable.

paulaOliverosHollis committed:
Fixed merge and correted whitespace.

paulaOliverosHollis committed:
FIxed WhiteSpave.
